### PR TITLE
Cleanup of old CUDA versions

### PR DESCRIPTION
**Pull request by mgorny**

### CODE REVIEW

**Comment by mgorny** [comment]
@gentoo/qa, @gentoo/sci.

**Comment by SoapGentoo** [comment]
@marbre can we also p.mask the affected versions along?

**Comment by jlec** [comment]
Why dropping all those USE masks entries? THat doesn't make sense to me.

**Comment by mgorny** [comment]
> Why dropping all those USE masks entries? THat doesn't make sense to me.

Because new versions of CUDA do not support x86. And old versions require old gcc.

**Comment by mgorny** [comment]
Hrm. We also need to stabilize a newer CUDA toolkit for amd64. @jlec, sounds good?

**Comment by jlec** [comment]
> Hrm. We also need to stabilize a newer CUDA toolkit for amd64. @jlec, sounds good?
Sure, never had problems with newer cuda. @marbre what do you think?

**Comment by SoapGentoo** [comment]
@jlec @marbre Personally, I would prefer `use.stable.mask`'ing `USE=cuda`, as Cuda is a special feature, stablizing it requires special hardware (I don't own nvidia hardware and don't plan on doing so), the fact that the SDK is packaged a bit dodgily after all and the fact that our ATs are still kinda overburdened are playing catchup.

**Comment by mgorny** [comment]
@SoapGentoo, could you finish this for me? You'd need to drop magma to ~arch, and add package.use.mask for cpyrit[cuda] since it needs old gcc.

We also possibly need to look for packages enforcing old gcc via version check.

File: profiles/base/package.use.mask [hunk 1]
```diff
@@ -7,6 +7,10 @@
 # This file is only for generic masks. For arch-specific masks (i.e.
 # mask everywhere, unmask on arch/*) use arch/base.
 
+# David Seifert <soap@gentoo.org> (07 May 2017)
```
Comment (mgorny) [inline profiles/base/package.use.mask]: This goes to amd64. Otherwise, the unmask there will have precedence.

**Comment by mgorny** [comment]
@jlec, @marbre: please re-review since this became mostly CUDA-related.

**Comment by mgorny** [comment]
I'm not sure if we oughtn't have a separate mask for CUDA now. And possibly start thinking of removing the old version along with revdeps.

File: profiles/package.mask [hunk 2]
```diff
@@ -29,6 +29,19 @@
...
+# compiler. Packages are no longer tested against them, and may fail
+# to build. If you are still using an old GCC version as the system
+# compiler, you should upgrade sys-devel/gcc and switch gcc-config
+# to a new compiler ASAP. If you only need those GCC versions for
+# specific local uses, unmask them freely.
+<sys-devel/gcc-4.9
+<dev-python/pycuda-2016
+<dev-util/nvidia-cuda-sdk-7
+<dev-util/nvidia-cuda-toolkit-7
+net-wireless/cpyrit-cuda
```
Comment (tamiko) [inline profiles/package.mask]: Can we please split this into two separate masks?

Further I would prefer to let the masking business of `sys-devel/gcc` to toolchain. :-)

**Comment by marbre** [comment]
@mgorny I have a paper deadline at the moment, but I'll check ASAP.

**Comment by mgorny** [comment]
One more thing: `media-gfx/nvidia-texture-tools[cuda]` requires very old gcc. We probably need to mask that too.

**Comment by mgorny** [comment]
@marbre, gentle reminder.

**Comment by marbre** [comment]
@mgrony Paper finished, back from a conference ;)

Actually, I somehow did not got to point why the cuda 6.5* versions were dropped to ~arch. Did I got it right that it is not intended to stabilize any newer cuda version and instead all cuda use flags needs to be masked?
It is correct that testing is somehow an issue, as executing tests isn't possible without an cuda capable GPU. However, compiling/installing packages normally isn't an issue.

**Comment by mgorny** [comment]
@SoapGentoo is the one playing that game. AFAIU the goal is to keep most of the sci-* related stuff in ~arch since we can't deal with stabilizing it properly. And yes, I think it's a problem to request arch teams to stabilize packages which they can't test properly.

**Comment by SoapGentoo** [comment]
@marbre is there an issue with removing stable keywords? sci is a mess as is, and destabling it allows us to be more agile in how we proceed and fix stuff.

**Comment by marbre** [comment]
@SoapGentoo I am fine with masking cuda, just wanted to get to know the background properly to not break anything in the future.

Actually, the changes look fine to my, except it is not clear wyhy cuda < 7 is added to `profiles/package.mask`. Is the intention to remove the 6.* versions as only gcc < 5 is supported and otherwise we get into trouble with the toolchain? These are the last releases supporting x86! However, I am fine with dropping x86 support.

**Comment by mgorny** [comment]
Yes, the whole affair started because I wanted to mask old gcc versions and that caused impossible deps for CUDA. So I suppose nobody object to merging this? Do we need to make any special announcement?

**Comment by jlec** [comment]
Finally! Good work everyone.

**Pull request closed without merging**